test/any/FetchSpecRunner-test.js

```diff
@@ -0,0 +1,186 @@
+    foam.register(MockHTTPRequest, 'foam.net.RetryHTTPRequest');
+  });
+
+  it('should send an error if invalid arguments are received as a message', function() {
+    var outputBox = ResultBox.create();
+    var errorBox = ResultBox.create();
+    var runner = FetchSpecRunner.create({
+      outputBox: outputBox,
+      errorBox: errorBox,
+    });
+
+    // Sending any object other than a PipelineMessage object should
+    // result in a message sent to errorBox.
+    var wrongObj = {};
+    runner.run(wrongObj);
+    expect(errorBox.results.length).toBe(1);
+
+    // Sending a PipelineMessage that is missing urls field should
+    // result in a message sent to errorBox.
+    var msg = PipelineMessage.create();
+    runner.run(msg);
+    expect(errorBox.results.length).toBe(2);
+  });
+
+  it('should spawn a worker and fetch the given URLs', function(done) {
+    var outputBox = ResultBox.create();
+    var errorBox = ResultBox.create();
+    var runner = FetchSpecRunner.create({
+      outputBox: outputBox,
+      errorBox: errorBox,
+    });
+
+    var urls = [
+      'https://html.spec.whatwg.org/multipage/common-microsyntaxes.html',
+      'https://html.spec.whatwg.org/multipage/urls-and-fetching.html',
+    ];
+    var msg = PipelineMessage.create({ urls: urls });
+    runner.run(msg);
+
+    // Check back in a second to allow for fetching.
+    setTimeout(function() {
+      expect(outputBox.results.length).toBe(2);
+      expect(errorBox.results.length).toBe(0);
+
+      var htmlFiles = outputBox.results.map(function(o) {
+        return o.object.htmlFile;
+      });
+
+      // Verify that correct pages were fetched.
+      expect(htmlFiles[0].id[0]).not.toBe(htmlFiles[1].id[0]);
+      expect(htmlFiles[0].contents).not.toBe(htmlFiles[1].contents);
+      done();
+    }, 1000);
+  });
+
+  it('should only fetch URLs once', function(done) {
+    var outputBox = ResultBox.create();
+    var errorBox = ResultBox.create();
+    var runner = FetchSpecRunner.create({
+      outputBox: outputBox,
+      errorBox: errorBox,
+    });
+
+    var urls = [
+      'https://html.spec.whatwg.org/multipage/common-microsyntaxes.html',
+      'https://html.spec.whatwg.org/multipage/common-microsyntaxes.html',
+    ];
+    var msg = PipelineMessage.create({ urls: urls });
+    runner.run(msg);
+
+    // Check back in a second to allow for fetching.
+    setTimeout(function() {
+      expect(outputBox.results.length).toBe(1);
+      expect(errorBox.results.length).toBe(0);
+      done();
+    }, 1000);
+  });
+
+  it('should send an error to errorBox if given a non-existent URL', function(done) {
+    var outputBox = ResultBox.create();
+    var errorBox = ResultBox.create();
+    var runner = FetchSpecRunner.create({
+      outputBox: outputBox,
+      errorBox: errorBox,
+    });
+
+    var urls = [
+      'https://w3c.github.io/mediacapture-imge', // URL has a typo, expecting 404.
```

Comment:
Now that you are mocking the requests, the typo isn't needed. Maybe define a constant for a URL that will produce a 404.